modules/dxp/apps/segments/segments-asah-connector/src/main/java/com/liferay/segments/asah/connector/internal/messaging/InterestTermsChecker.java

```diff
@@ -75,19 +75,15 @@ public void checkInterestTerms(long companyId, String userId) {
 			return;
 		}
 
-		Stream<Topic> stream = topics.stream();
+		List<String> termList = new ArrayList<>();
 
-		String[] terms = stream.flatMap(
-			topic -> {
-				List<Topic.TopicTerm> topicTerms = topic.getTerms();
-
-				return topicTerms.stream();
+		for (Topic topic : topics) {
+			for (Topic.TopicTerm term : topic.getTerms()) {
+				termList.add(term.getKeyword());
 			}
-		).map(
```

Comment:
@W2jx, for the second loop, you may use termList.addAll(TransformUtil.transform(topic.getTerms(), term -> term.getKeyword())) to replace.